subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultDependencyHandler.java

```diff
@@ -123,6 +125,11 @@ public Dependency create(Object dependencyNotation, @Nullable Closure configureC
     @SuppressWarnings("rawtypes")
     private Dependency doAdd(Configuration configuration, Object dependencyNotation, @Nullable Closure configureClosure) {
         if (dependencyNotation instanceof Configuration) {
+            DeprecationLogger.deprecateBehaviour("Adding a Configuration as a dependency is a confusing behavior which isn't recommended.")
+                .withAdvice("You should use Configuration#extendsFrom instead.")
+                .willBeRemovedInGradle8()
```

Comment:
I still think we can do this for 7.0 given this will be part of 6.7

Reply:
I think it's too late for removal in 7.0. A couple releases isn't enough for most users.